### ISSUE TITLE
mlpack_krann has no tests

### ISSUE THREAD

**Issue by rcurtin** [mlpack/mlpack#2432]
#### Issue description

The `mlpack_krann` binding, found in `src/mlpack/methods/rann/krann_main.cpp`, doesn't have a corresponding test in `src/mlpack/tests/main_tests/`.  This leads to errors like those that are fixed in #2431 .

#### Your environment

 * version of mlpack: git master
 * operating system: debian
 * compiler: gcc 9
 * version of dependencies (Boost/Armadillo): shouldn't matter for this issue
 * any other environment information you think is relevant:

#### Steps to reproduce

Build tests, try to run `bin/mlpack_test -t KRANNMainTest`, see that it doesn't exist. :)

#### Expected behavior

`bin/mlpack_test -t KRANNMainTest` should run some integration tests on the `mlpack_krann` binding.

#### Actual behavior

```
$ bin/mlpack_test -t KRANNMainTest
Test setup error: no test cases matching filter or all test cases were disabled
```

#### How to fix

Basically, we just need to implement tests for the `mlpack_krann` binding.  There is a lot of information for how to do that in #1152.

**Comment by UtR491** [comment]
Is anybody working on this one? I think I understand the issue, would like to give it a try.

**Comment by jeffin143** [comment]
@UtR491 go ahead please :)

**Comment by UtR491** [comment]
So sorry for being so late. I had some homework to submit. I think I have run into a problem, I wrote 11 tests for alpha, tau, dimensions and so on. I built the project, initially there were some errors, but I corrected them, now the test just runs indefinitely. I read some boost documentation but couldn't figure out where the problem was.[The code](https://github.com/UtR491/mlpack/blob/krann-test/src/mlpack/tests/main_tests/krann_test.cpp)
